Source/Persistence Storage/PurchaseStorage.swift

```diff
@@ -2,6 +2,22 @@ public protocol PurchaseStorage : AnyObject {
     func record(forProductIdentifier productIdentifier: String) -> PurchaseRecord?
     func save(_ record: PurchaseRecord) -> PurchaseStorageUpdateResult
     func removeRecord(forProductIdentifier productIdentifier: String) -> PurchaseStorageUpdateResult
+    func state(for product: Product) -> PurchasedState
+}
+
+public extension PurchaseStorage {
+  /// Returns the state for a `product`. Consumable products always report that they are `notPurchased`.
+  public func state(for product: Product) -> PurchasedState {
```

Comment:
self review: rest of the purchase storage api uses product identifiers. probably a good idea to follow suit here…